File: LC/Easy/121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        curr_min = 0
        ans = 0
    
        for i in range(1, len(prices)):
            if (prices[curr_min] > prices[i]):
                curr_min = i
            ans = max(ans, prices[i] - prices[curr_min])    
        
        
        return ans

File: LC/C++/Medium/121.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans = 0;
        int minPrice = prices[0];

        for (int n : prices) {
            minPrice = min(minPrice, n);
            ans = max(ans, n - minPrice);
        }

        return ans;
    }
};